lib/Common/CommonDefines.h

```diff
@@ -140,7 +140,11 @@
 #define ENABLE_JS_ETW                               // ETW support
 #else
 #define SYSINFO_IMAGE_BASE_AVAILABLE 0
+#ifndef ENABLE_VALGRIND
 #define ENABLE_CONCURRENT_GC 1
+#else
+#define ENABLE_CONCURRENT_GC 0
```

Comment:
X86 argument makes sense and will apply. However as given from the description 32GB approach doesn't work because we don't have control over stack ptr addresses. Will start an offline discussion on the second one. We might have a bug with write barrier.